src/rust/receiver-mock/src/router/otlp.rs

```diff
@@ -0,0 +1,296 @@
+                .app_data(get_default_app_data())
+                .service(web::scope("/v1").route("/logs", web::post().to(handler_receiver_otlp_logs)))
+                .default_service(web::get().to(handler_receiver)),
+        )
+        .await;
+
+        let request = test::TestRequest::post()
+            .uri("/v1/logs")
+            .insert_header(("Content-Type", OTLP_PROTOBUF_FORMAT_CONTENT_TYPE))
+            .to_request();
+
+        let response = test::call_service(&mut app, request).await;
+        assert_eq!(response.status(), StatusCode::INTERNAL_SERVER_ERROR);
+        let body = response.into_body();
+        assert_eq!(
+            get_body_str(body),
+            format!("Dropping data for {}", OTLP_PROTOBUF_FORMAT_CONTENT_TYPE)
+        );
+    }
+
+    #[actix_rt::test]
+    async fn otlp_logs_unrelated_content_type_test() {
+        let content_type = "unrelated/type";
+        let opts = get_default_options();
+        let mut app = test::init_service(
+            App::new()
+                .app_data(web::Data::new(opts.clone()))
+                .app_data(get_default_app_data())
+                .service(web::scope("/v1").route("/logs", web::post().to(handler_receiver_otlp_logs)))
+                .default_service(web::get().to(handler_receiver)),
+        )
+        .await;
+
+        let request = test::TestRequest::post()
+            .uri("/v1/logs")
+            .insert_header(("Content-Type", content_type))
+            .to_request();
+
+        let response = test::call_service(&mut app, request).await;
+        assert_eq!(response.status(), StatusCode::BAD_REQUEST);
+    }
+
+    #[actix_rt::test]
+    async fn otlp_logs_store_test() {
+        let opts = get_default_options();
+        let mut app = test::init_service(
+            App::new()
+                .app_data(web::Data::new(opts.clone()))
+                .app_data(get_default_app_data())
+                .service(web::scope("/v1").route("/logs", web::post().to(handler_receiver_otlp_logs)))
+                .route("/logs/count", web::get().to(handler_logs_count))
+                .default_service(web::get().to(handler_receiver)),
+        )
+        .await;
+
+        {
+            let request = test::TestRequest::post()
+                .uri("/v1/logs")
+                .insert_header(("Content-Type", OTLP_PROTOBUF_FORMAT_CONTENT_TYPE))
+                .set_payload(get_sample_request_body())
+                .to_request();
+
+            let response = test::call_service(&mut app, request).await;
+            assert_eq!(response.status(), StatusCode::OK);
+        }
+
+        // count all the logs
+        {
+            let req = test::TestRequest::get().uri("/logs/count").to_request();
+            let resp = test::call_service(&mut app, req).await;
+
+            let response_body: LogsCountResponse = test::read_body_json(resp).await;
+
+            assert_eq!(response_body.count, 2);
+        }
+    }
```

Comment:
How about some unit tests for the pure conversion functions? Should be much simpler than these E2E tests.